Automattic/jetpack
# Launchpad: Add "Enable subscribers modal" task to Newsletter task lists

Fixes Automattic/wp-calypso#79560

## Proposed changes:

* Add a new task "Enable subscribers modal" to the Newsletter Paid and Newsletter Free task lists.
* Task is only shown if the filter `jetpack_subscriptions_modal_enabled` returns true.
* Add function hooked into `update_option_sm_enabled` to mark the task as complete when the user changes the value to true.
* Add `sm_enabled` site option to the Jetpack Sync allowed options array so this will work on Atomic sites.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






* Apply this patch to your sandbox and sandbox the API
* Go to the developer console and GET WP REST API `wpcom/v2/sites/siteSlug/launchpad?checklist_slug=intent-free-newsletter` and `wpcom/v2/sites/siteSlug/launchpad?checklist_slug=intent-paid-newsletter`
* You should see the `enable_subscribers_modal` task in both lists in the results:

<img width="1043" alt="Screenshot 2023-07-24 at 12 42 14 PM" src="https://github.com/Automattic/jetpack/assets/2124984/6562e383-6930-440c-a88d-5098f8f7d412">
